src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
I think we have a common library to deal with semaphores/mutex. Maybe should move this (and the mutex management in general) to that library. In addition, we have a feature to capture statistics about how many time threads are waiting in each mutex and probably this new mutext should be also included there.

@kzangeli could provide more detail or particular changes suggestions, he has designed and implemented that part.